posydon/utils/common_functions.py

```diff
@@ -2578,6 +2578,68 @@ def calculate_binding_energy(donor_mass, donor_radius, donor_dm,
         print("Ebind = Grav_energy + factor_internal_energy*U_i  :  ", Ebind_i)
     return Ebind_i
 
+def calculate_Mejected_for_integrated_binding_energy(profile, Ebind_threshold,
+                             mc1_i, rc1_i,
+                             m1_i = 0.0, radius1 = 0.0,
+                             factor_internal_energy=1.0,tolerance=0.001
+                             ):
+    """Calculate the mass lost from the envelope for an energy budget of Ebind_threshold
+
+    Parameters
+    ----------
+    profile : numpy.array
+        Donor's star profile from MESA
+    Ebind_threshold : float
+        Orbital energy used from the spiral in to partial unbind the envelope. Positive
+        We integrate from surface to calcualte the partial loss of mass during CE that merges.
+    factor_internal_energy : float
+        The factor to multiply with internal energy to be taken into
+        account when we calculate the  binding energy of the enevelope
+    verbose : bool
+        In case we want information about the CEE  (the default is False).
+
+    Returns
+    -------
+    Ebind_i : float
+        The total binding energy of the envelope of the star
+
+    """
+
+    donor_mass, donor_radius, donor_dm = get_mass_radius_dm_from_profile(
+        profile, m1_i, radius1, tolerance)
+    specific_internal_energy = get_internal_energy_from_profile(
+        common_envelope_option_for_lambda = "lambda_from_profile_gravitational_plus_internal_minus_recombination",
+        profile = profile, tolerance = tolerance)
+
+    # Sum of gravitational energy from surface towards inside
+    Grav_energy = 0.0
+    # Sum of internal energy from surface towards.
+    U_energy = 0.0
+    # sum from surface to the core. Your threshold is in element [ind_threshold]
+    # in a normal MESA (and POSYDON) profile
+    i = 0
+    Ebind_so_far = 0.0 # the integration from surface going inwards of the binding energy (negative in principle)
+
+    while abs(Ebind_so_far) < Ebind_threshold:
```

Comment:
I'd suggest to add here a check on `i` to make sure it will never exceed the number of points in the profiles.

Reply:
Done